### PR TITLE
Use topic exchange consumers

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: src/modules/users/consumers/PermissionConsumer.ts [hunk 1]
```diff
@@ -40,7 +44,9 @@ export class PermissionConsumer extends AbstractAmqpQueueConsumer<
       },
       {
         creationConfig: {
+          exchange: PermissionConsumer.EXCHANGE_NAME,
           queueName: PermissionConsumer.QUEUE_NAME,
+          topicPattern: '',
```
Comment (kibertoad) [inline src/modules/users/consumers/PermissionConsumer.ts:49]: @CarlosGamero Should we make this parameter optional and default to ''? It is mandatory for queue binding operation in RabbitMQ, but empty value is acceptable.

File: src/modules/users/consumers/PermissionsConsumer.spec.ts [hunk 2]
```diff
@@ -88,14 +88,20 @@ describe('PermissionsConsumer', () => {
...
+          payload: {
+            userIds,
+            permissions: perms,
+          },
+          type: 'permissions.added',
         },
-        type: 'permissions.added',
-      })
+        {
+          routingKey: '',
```
Comment (kibertoad) [inline src/modules/users/consumers/PermissionsConsumer.spec.ts:102]: @CarlosGamero Should we make this parameter optional and default to ''? It is mandatory for publish to topic operation in RabbitMQ, but empty value is acceptable.

File: src/modules/users/consumers/PermissionConsumer.ts (hunk 1 above)
Comment (CarlosGamero) [reply src/modules/users/consumers/PermissionConsumer.ts:49]: I am not sure what are the implications of having it empty, but if it is a valid value the can be use in "real world" cases I would vote for having it optional to simplify the API

File: src/modules/users/consumers/PermissionsConsumer.spec.ts (hunk 2 above)
Comment (CarlosGamero) [reply src/modules/users/consumers/PermissionsConsumer.spec.ts:102]: I would say the same as before, if it is a valid value used in real cases I would vote for simplify the API by having it optional

File: src/modules/users/consumers/PermissionConsumer.ts (hunk 1 above)
Comment (kibertoad) [reply src/modules/users/consumers/PermissionConsumer.ts:49]: empty is treated same as any other value, so you will get messages if both consumer and publisher are configured to use empty topic. this means that routing will be done on a purely exchange/queue level.

Comment (CarlosGamero) [reply src/modules/users/consumers/PermissionConsumer.ts:49]: I see, thanks for explaining! 🙏 So I agree with having it optional 🙌

**Pull request merged**